### PR TITLE
Throw exception on missing configmap during initial load (fixes #88)

### LINKED ISSUE #88: Throw exception on missing Configmap?
We've stumbled upon an issue where we didn't see that a configmap was not loaded correctly because of a typo and our logging level being set to Information+. It would be ideal if the library could throw an exception, seeing as missing ConfigMaps would be a major issue for most applications.

Would this be okay with you and if so, should we open a PR?

Referencing this line:
https://github.com/tintoy/dotnet-kube-client/blob/e3d429ec9fcf185f6709613c32ad7846893522ba/src/KubeClient.Extensions.Configuration/ConfigMapConfigurationProvider.cs#L145

### PR DESCRIPTION
**Pull request by PhilippCh**

### CODE REVIEW

**Comment by tintoy** [comment]
Hi - thanks for contributing! Are you able to run the tests locally? The CI test run seems to be hanging on this branch...

**Comment by tintoy** [comment]
PS. Sorry to leave you hanging but I’m about to go to bed (I’m in the AU time zone), will pick this up first thing tomorrow morning :)

**Comment by PhilippCh** [comment]
Unit tests seem to be all-green locally. I think Travis completed just now as well.
No worries, let me know if the PR is okay like that or if I should revise something :)

File: src/KubeClient.Extensions.Configuration/ConfigMapConfigurationProvider.cs [hunk 1]
```diff
@@ -142,9 +144,10 @@ void Load(ConfigMapV1 configMap)
             }
             else
             {
-                Log.LogTrace("ConfigMap {ConfigMapName} was not found in namespace {KubeNamespace}.", _configMapName, _kubeNamespace ?? _client.DefaultNamespace);
-
                 Data = new Dictionary<string, string>();
+                
+                Log.LogTrace("ConfigMap {ConfigMapName} was not found in namespace {KubeNamespace}.", _configMapName, _kubeNamespace ?? _client.DefaultNamespace);
+                throw new KubeClientException($"ConfigMap {_configMapName} was not found in namespace {_kubeNamespace}.");
```
Comment (tintoy) [inline src/KubeClient.Extensions.Configuration/ConfigMapConfigurationProvider.cs]: I think you might need to check if `throwOnNotFound` is `true` before throwing?

**Comment by tintoy** [comment]
I've added a [small tweak](https://github.com/tintoy/dotnet-kube-client/pull/89/commits/eb0a84532d6ba167ba687ce9f66890aea52b0ce7) to to your changes - can you have a look over it and let me know if you're happy with that? Hopefully it's still doing what you wanted :)

**Comment by PhilippCh** [comment]
Thanks a lot, that's still working for me! Sorry I didn't get back to you sooner, I couldn't find the time the last couple days.

**Comment by tintoy** [comment]
No problem :)

Will merge and publish this afternoon.

**Comment by tintoy** [comment]
Sorry for the delay - been a busy week!

**Comment by PhilippCh** [comment]
Thanks a lot for merging! Would you mind publishing to NuGet as well? Do you have a release schedule this could fit in?

**Comment by tintoy** [comment]
Published [v2.2.15](https://travis-ci.org/tintoy/dotnet-kube-client/builds/568829536) (may take a while for packages to be indexed on NuGet) 😀

**Comment by PhilippCh** [comment]
Thank you!

**Pull request merged**